### PR TITLE
Add support for payment_intent on invoice and default_payment_method

### PR DESCRIPTION
**Pull request by remi-stripe**
This is now ready to be released. Those fields will be public later this week and are stable. Having this out ahead of time is likely the best approach.

r? @mickjermsurawong-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: customer.go [hunk 1]
```diff
@@ -53,8 +53,9 @@ type CustomerInvoiceCustomFieldParams struct {
 // CustomerInvoiceSettingsParams is the structure containing the default settings for invoices
 // associated with this customer.
 type CustomerInvoiceSettingsParams struct {
-	CustomFields []*CustomerInvoiceCustomFieldParams `form:"custom_fields"`
-	Footer       *string                             `form:"footer"`
+	CustomFields         []*CustomerInvoiceCustomFieldParams `form:"custom_fields"`
+	DefaultPaymentMethod *string                             `form:"default_payment_method"`
+	Footer               *string                             `form:"footer"`
```
Comment (mickjermsurawong-stripe) [inline customer.go]: Ah just flagging that `default_payment_method` is only present in update and not create.

File: invoice.go [hunk 2]
```diff
@@ -87,6 +87,7 @@ type InvoiceParams struct {
 	CustomFields         []*InvoiceCustomFieldParams `form:"custom_fields"`
 	Customer             *string                     `form:"customer"`
 	DaysUntilDue         *int64                      `form:"days_until_due"`
+	DefaultPaymentMethod *string                     `form:"default_payment_method"`
```
Comment (mickjermsurawong-stripe) [inline invoice.go]: @remi-stripe 
We miss `payment_method` in  `InvoicePayParams`

File: customer.go [hunk 3]
```diff
@@ -110,6 +111,20 @@ type Customer struct {
...
+}
+
+// CustomerInvoiceSettings is the structure containing the default settings for invoices associated
+// with this customer.
+type CustomerInvoiceSettings struct {
+	CustomFields         []*CustomerInvoiceCustomField `json:"custom_fields"`
+	DefaultPaymentMethod *PaymentMethod                `json:"default_payment_method"`
+	Footer               string                        `json:"footer"`
+}
+
```
Comment (mickjermsurawong-stripe) [inline customer.go]: We miss expandable `default_payment_method` on Customer struct here.
Or do you leave it out intentionally?

**Comment by mickjermsurawong-stripe** [comment]
ptal @remi-stripe

File: customer.go (hunk 3 above)
Comment (mickjermsurawong-stripe) [reply customer.go]: Oh whoops my bad. It is correct here!

File: customer.go (hunk 1 above)
Comment (remi-stripe) [reply customer.go]: Fairly sure this is a bug in the backend so will flag.

**Comment by remi-stripe** [comment]
@mickjermsurawong-stripe Fixed the Pay one, PTAL

**Comment by mickjermsurawong-stripe** [comment]
LGTM
thanks @remi-stripe

**Pull request merged**